azure/services/vnetpeerings/client.go

```diff
@@ -61,36 +68,135 @@ func (ac *AzureClient) Get(ctx context.Context, resourceGroupName, vnetName, pee
 	return ac.peerings.Get(ctx, resourceGroupName, vnetName, peeringName)
 }
 
-// CreateOrUpdate creates or updates a virtual network peering in the specified virtual network.
-func (ac *AzureClient) CreateOrUpdate(ctx context.Context, resourceGroupName, vnetName, peeringName string, peering network.VirtualNetworkPeering) error {
-	ctx, span := tele.Tracer().Start(ctx, "vnetpeerings.AzureClient.CreateOrUpdate")
+// CreateOrUpdateAsync creates or updates a virtual network peering asynchronously.
+// It sends a PUT request to Azure and if accepted without error, the func will return a Future which can be used to track the ongoing
+// progress of the operation.
+func (ac *AzureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, azureautorest.FutureAPI, error) {
+	ctx, span := tele.Tracer().Start(ctx, "routetables.AzureClient.CreateOrUpdateAsync")
 	defer span.End()
 
-	future, err := ac.peerings.CreateOrUpdate(ctx, resourceGroupName, vnetName, peeringName, peering, network.SyncRemoteAddressSpaceTrue)
+	peering, err := ac.VnetPeeringParams(ctx, spec)
 	if err != nil {
-		return err
+		return nil, nil, errors.Wrapf(err, "failed to get desired parameters for virtual network peering %s", spec.ResourceName())
+	} else if peering == nil {
+		// nothing to do here.
+		return nil, nil, nil
```

Comment:
I think we'll also want to do https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1837/commits/aee609ef6e1268d28b6f9a49c6b56a9cb9888875 after I figure out why the tests are failing in that pr